addons/mail/static/src/core/common/store_service.js

```diff
@@ -428,7 +428,7 @@ export class Store extends BaseStore {
             return;
         }
         let chat = partner.searchChat();
-        if (!chat || !chat.is_pinned) {
+        if (!chat || !chat.selfMember?.is_pinned) {
             chat = await this.joinChat(partner.id);
```

Comment:
We have to be careful using the optional chaining and negative condition. In this case, the condition will be true if chat.selfMember is `undefined`.
This was not the case before so we might want to check that `this.selfMember` exists.